flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/snapshot/RocksNativeFullSnapshotStrategy.java

```diff
@@ -219,34 +219,40 @@ public SnapshotResult<KeyedStateHandle> get(CloseableRegistry snapshotCloseableR
             }
         }
 
-        private void uploadSstFiles(
-                @Nonnull Map<StateHandleID, StreamStateHandle> privateFiles,
+        /** upload files and return total uploaded size. */
+        private long uploadSnapshotFiles(
+                @Nonnull List<HandleAndLocalPath> privateFiles,
                 @Nonnull CloseableRegistry snapshotCloseableRegistry,
                 @Nonnull CloseableRegistry tmpResourcesRegistry)
                 throws Exception {
 
             // write state data
             Preconditions.checkState(localBackupDirectory.exists());
 
-            Map<StateHandleID, Path> privateFilePaths = new HashMap<>();
-
             Path[] files = localBackupDirectory.listDirectory();
+            long uploadedSize = 0;
             if (files != null) {
+                List<Path> privateFilePaths = new ArrayList<>(files.length);
+
                 // all sst files are private in full snapshot
                 for (Path filePath : files) {
-                    final String fileName = filePath.getFileName().toString();
-                    final StateHandleID stateHandleID = new StateHandleID(fileName);
-                    privateFilePaths.put(stateHandleID, filePath);
+                    privateFilePaths.add(filePath);
```

Comment:
Is this intermediate variable still necessary? Can't we just use `Arrays.asList(files)`?